tasks/Task.cpp

```diff
@@ -0,0 +1,189 @@
+    mDriver = std::unique_ptr<gps_ublox::Driver>(new Driver());
+    if (!_io_port.get().empty())
+        mDriver->openURI(_io_port.get());
+    setDriver(mDriver.get());
+
+    if (! TaskBase::configureHook())
+        return false;
+
+    loadConfiguration();
+
+    guard.commit();
+    return true;
+}
+bool Task::startHook()
+{
+    if (! TaskBase::startHook())
+        return false;
+    return true;
+}
+void Task::updateHook()
+{
+    TaskBase::updateHook();
+}
+
+template<typename T>
+T may_invalidate(T const& value)
+{
+    if (value == T::Zero())
+        return T::Ones() * base::unknown<double>();
+    else return value;
+}
+base::samples::RigidBodyState Task::convertToRBS(const gps_ublox::GPSData &data) const {
+    base::samples::RigidBodyState rbs;
+    gps_base::Solution geodeticPosition;
+
+    Eigen::Vector3d body2ned_velocity = Eigen::Vector3d(
+        data.vel_ned.x(), data.vel_ned.y(), data.vel_ned.z());
+
+    rbs.time = data.time;
+    rbs.velocity = Eigen::AngleAxisd(
+        M_PI, Eigen::Vector3d::UnitX()) * may_invalidate(body2ned_velocity);
+
+    geodeticPosition.latitude = data.latitude.getDeg();
+    geodeticPosition.longitude = data.longitude.getDeg();
+    geodeticPosition.altitude = data.height;
+    geodeticPosition.deviationAltitude = data.vertical_accuracy;
+    geodeticPosition.deviationLatitude = data.horizontal_accuracy;
+    geodeticPosition.deviationLongitude = data.horizontal_accuracy;
+
+    base::samples::RigidBodyState nwu = mUTMConverter.convertToNWU(geodeticPosition);
+    rbs.position = nwu.position;
+    rbs.cov_position = nwu.cov_position;
+    return rbs;
+}
+gps_base::SatelliteInfo Task::convertToBaseSatelliteInfo(const SatelliteInfo &sat_info) const
+{
+    gps_base::SatelliteInfo rock_sat_info;
+
+    rock_sat_info.time = base::Time::now();
+    for (const SatelliteInfo::Data &data : sat_info.signals) {
+        gps_base::Satellite sat;
+        sat.azimuth = (int)data.azimuth.getDeg();
+        sat.elevation = (int)data.elevation.getDeg();
+        sat.PRN = data.satellite_id;
+        sat.SNR = (double)data.signal_strength;
+
+        rock_sat_info.knownSatellites.push_back(sat);
+    }
+    return rock_sat_info;
+}
+gps_base::Solution Task::convertToBaseSolution(const GPSData &data) const
+{
+    gps_base::Solution solution;
+
+    // solution.ageOfDifferentialCorrections = <unavailable>
+    // solution.geoidalSeparation = <unavailable>
+    solution.altitude = data.height_above_mean_sea_level;
+    solution.deviationAltitude = data.vertical_accuracy;
+    solution.deviationLatitude = data.horizontal_accuracy;
+    solution.deviationLongitude = data.horizontal_accuracy;
+    solution.noOfSatellites = data.num_sats;
+    switch (data.fix_type) {
+        case GPSData::NO_FIX:
+        case GPSData::TIME_ONLY:
+            solution.positionType = gps_base::NO_SOLUTION;
+            break;
+        case GPSData::DEAD_RECKONING:
+            solution.positionType = gps_base::RTK_FLOAT;
```

Comment:
RTK_FLOAT is most definitely not  dead reckoning. DEAD_RECKONING basically means that there is no solution, but that the unit has an internal kalman filter that allows it to integrate other data sources. There's no GNSS `positionType` for dead reckoning.